sassoftware/viya4-iac-azure
# (IAC-503) Remove connect class references from iac-azure

# Changes
Starting from 2021.2.6 the connect node pool is optional, removing it from the examples and defaults in variables.tf. 
sample-input-connect.tfvars was created to show the user how to create the node pool in case they are deploying on an older cadence.

# Tests

**Note**: The follow scenarios were run and produced the expected results, for more detail and the outputs please see the internal IAC-503 ticket.

- Use new IAC code, new install Viya 2021.2.6 and beyond: users use IAC/DAC default setting to deploy Viya 4, result in a successful deployment without "connect" node pool
- Use new IAC code, new install Viya 2021.2.6 and beyond: users are able to have a successful deployment with connect node pool (using the new sample-input-connect.tfvars)
- Use new IAC code (use the default settings), new install Viya 4 old cadences, e.g. 2021.2.4: This is a breaking change. This won’t work because connect pool is needed for 2021.2.5 and prior but defaults don’t have it. (Note: People can use old IAC code if they want to deploy 2021.2.5 and prior or see the next case).
- Use new IAC code (default settings), new install Viya 4 old cadences, e.g. 2021.2.5 (connect is needed for those cadences). Use the new example file sample-input-connect.tfvars and deploy 2021.2.5 will result in a successful deployment.
- Upgrade: Customer has Viya 2021.2.5 running and they complete a UIP to 2021.2.6. At this point, they have “connect”. The customer then updates their tfvars to remove reference to the connect node pool and asks Terraform to update the cluster. This results in a working Viya.
